### ISSUE TITLE
formatInline works strangely

### ISSUE THREAD

**Issue by marcghorayeb** [tiff/wysihtml5#221]
Hello,

I found that formatInline works erraticly depending on the way you apply tags.
For example:
1. Select a line, make it italic.
2. Select a couple of words inside that line, make them bold.
3. Select a word out of the words you just made bold, underline it.

The result is a clean HTML, with only one occurence of each tag.
However, do that the other way around, underlining one word, then bolding a couple of words, and then making the whole line italic, and you have a horrible mess of tags.

Basically, every time the formatInline is applied, it looks for tags inside it and will close and then reopen the format inside those tags.

I can reproduce it on the project's homepage so I'm guessing I'm not the only one with this problem?

**Comment by marcghorayeb** [comment]
Looking at the code, this behavior is "normal" based on the scenario number 3:
https://github.com/xing/wysihtml5/blob/master/dist/wysihtml5-0.3.0.js#L7264-7267

However, the scenario i'm describing is one where the selection goes beyond all the tags, and thus should fall in line with scenario number 2...

**Comment by tiff** [comment]
Can you describe your scenario like in the code (including what you would expect)? Thanks! It will help me to create a test case.

**Comment by marcghorayeb** [comment]
Hey, thanks for the quick response! Something like this:

```
 *   #7 unformatted text selected containing formatted text:
 *      ab|c <span>defgh</span> <span>ijk</span> lmnop|qr
 *   output:
 *      ab<b>|c <span>defgh</span> <span>ijk</span> lmnop</b>|qr
```

**Comment by marcghorayeb** [comment]
@tiff was that what you wanted?

**Comment by tiff** [comment]
Yep, thanks! I will look into this as soon as wysihtml5 0.4 is out. Sorry for not prioritizing high. If you wanna give it a look I'd be happy to support you.

**Comment by marcghorayeb** [comment]
hey @tiff , any idea when 0.4 will be out? :) didn't have a lot of time to look into this issue so far ...
